teuthology/results.py

```diff
@@ -36,21 +36,26 @@ def main(args):
 
 
 def results(archive_dir, name, email, timeout, dry_run):
-    archive_base = os.path.split(archive_dir)[0]
-    serializer = ResultsSerializer(archive_base)
     starttime = time.time()
 
     if timeout:
         log.info('Waiting up to %d seconds for tests to finish...', timeout)
-    while serializer.running_jobs_for_run(name) and timeout > 0:
+
+    reporter = ResultsReporter()
+    while timeout > 0:
```

Comment:
``timeout`` is a value that doesn't change. This block will only run if we want ``teuthology-results`` to block for a time waiting for the run to finish.